src/m2n/tests/PointToPointCommunicationTest.cpp

```diff
@@ -419,6 +419,276 @@ BOOST_AUTO_TEST_CASE(ConnectionTest, * testing::OnSize(4))
+        BOOST_TEST(mesh->vertices()[2].getCoords()[0]==5.50);
+        BOOST_TEST(mesh->vertices()[2].getCoords()[1]==0.0);
+        BOOST_TEST(mesh->vertices()[3].getCoords()[0]==1.0);
+        BOOST_TEST(mesh->vertices()[3].getCoords()[1]==2.0);        
+      }
+
+      if(utils::Parallel::getProcessRank() ==3 )
+      {      
+        BOOST_TEST(mesh->vertices().size()==4);
+        BOOST_TEST(mesh->vertices()[2].getCoords()[0]==1.50);
+        BOOST_TEST(mesh->vertices()[2].getCoords()[1]==0.0);
+        BOOST_TEST(mesh->vertices()[3].getCoords()[0]==1.50);
+        BOOST_TEST(mesh->vertices()[3].getCoords()[1]==2.0);
+      }
+    
+  }
+  
+  utils::MasterSlave::_communication = nullptr;
+  utils::MasterSlave::reset();
+  utils::Parallel::synchronizeProcesses();
+  utils::Parallel::clearGroups();
+  mesh::Mesh::resetGeometryIDsGlobally();
+  mesh::Data::resetDataCount();
+  utils::Parallel::setGlobalCommunicator(utils::Parallel::getCommunicatorWorld());
+}
+
+BOOST_AUTO_TEST_CASE(P2PComLCMTest, * testing::OnSize(4))
+{
+  
+  assertion(utils::Parallel::getCommunicatorSize() == 4);
+
+//  com::PtrCommunicationFactory cf(new com::SocketCommunicationFactory);
+  com::PtrCommunicationFactory cf(new com::MPIPortsCommunicationFactory);
+  
+  utils::MasterSlave::_communication = std::make_shared<com::MPIDirectCommunication>();
+
+  mesh::PtrMesh mesh(new mesh::Mesh("Mesh", 2, true));
+  std::map<int, std::vector<int>> localCommunicationMap;
+
+  switch (utils::Parallel::getProcessRank()) {
+  case 0: {
+    utils::Parallel::splitCommunicator("Fluid.Master");
+
+    utils::MasterSlave::_rank       = 0;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = true;
+    utils::MasterSlave::_slaveMode  = false;
+
+    utils::MasterSlave::_communication->acceptConnection("Fluid.Master", "Fluid.Slave", utils::Parallel::getProcessRank());
+    utils::MasterSlave::_communication->setRankOffset(1);   
+
+    mesh->getConnectedRanks().push_back(0);    
+    localCommunicationMap[0].push_back(102);
+    localCommunicationMap[0].push_back(1022);
+    localCommunicationMap[0].push_back(10222);
+    localCommunicationMap[1].push_back(103);
+    localCommunicationMap[1].push_back(1033);
+    localCommunicationMap[1].push_back(10333);
+   
+    break;
+  }
+  case 1: {
+    utils::Parallel::splitCommunicator("Fluid.Slave");
+
+    utils::MasterSlave::_rank       = 1;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = false;
+    utils::MasterSlave::_slaveMode  = true;
+
+    utils::MasterSlave::_communication->requestConnection("Fluid.Master", "Fluid.Slave", 0, 1);
+    
+    mesh->getConnectedRanks().push_back(1);    
+    localCommunicationMap[0].push_back(112);
+    localCommunicationMap[0].push_back(1122);
+    localCommunicationMap[0].push_back(11222);
+    localCommunicationMap[1].push_back(113);
+    localCommunicationMap[1].push_back(1133);
+    localCommunicationMap[1].push_back(11333);
```

Comment:
Is `11333` some sort of decimal encoding or do you really want to use a vertex ID of 11333?

Reply:
Yes, this is what I meant by "encoding". Can you please document briefly?

Reply:
done!